awscli/customizations/configure/addmodel.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+from botocore.model import ServiceModel
+
+from awscli.customizations.commands import BasicCommand
+
+
+def _get_endpoint_prefix_to_name_mappings(session):
+    # Get the mappings of endpoint prefixes to service names from the
+    # available service models.
+    prefixes_to_services = {}
+    for service_name in session.get_available_services():
+        service_model = session.get_service_model(service_name)
```

Comment:
I get that running this command is typically a one-off, but it seems excessive to load *every* single service JSON model just to copy a service model to a directory.  I understand why this was added but can we not just require the special cases require a `--service-name` argument?

I would think the "real" fix would be capturing the botocore specific renames in a JSON file somewhere in botocore.  This isn't the only place where this information was needed.